### ISSUE TITLE
PR Discussion: chartpress --reset on the Chart.yaml

### ISSUE THREAD

**Issue by consideRatio** [jupyterhub/chartpress#44]
I want a run of `chartpress --reset` to make someone that has developed in z2jh/binderhub locally be able to run `git add .` and then `git commit` instead of ending up to first clean up the changes of using `chartpress` to develop locally just because chartpress has modified Chart.yaml and values.yaml.

When we run `chartpress --reset`, we modify `Chart.yaml` and `values.yaml`. The configuration option `resetTag` that defaults to `set-by-chartpress` is only used within `values.yaml` while in `Chart.yaml` another logic is used. The logic [is simple](https://github.com/jupyterhub/chartpress/blob/9ded86b3aa9f577d90ae14fe97c8b75c8cbc2b87/chartpress.py#L309-L310):

```python
        if reset:
            version = chart['version'].split('-')[0]
```

In other words, it will take something like 0.9-asdf123 and make it 0.9. We currently have `0.9-dev`. I think we should simply write `0.9` or specifically 0.9.0. Or, make chartpress append `-dev` like we do in z2jh (but [not in binderhub's Helm chart's Chart.yaml](https://github.com/jupyterhub/binderhub/blob/a168d069772012c52f9ac7056ec22d779927ae69/helm-chart/binderhub/Chart.yaml#L1-L4)).

It could be nice to reset to `x.y.z-dev` to make it clear that x.y.z isn't released yet.

**Comment by consideRatio** [comment]
## Implementation strategy discussion

One could consider using another strategy, that would allow the choice to be opinionated to remain. The strategy would be to use `git show HEAD:jupyterhub/<file>` of the files we reset and inspect what their past values were instead.

If you would like to do this way manually, you would perhaps want to use `[git checkout --patch](https://git-scm.com/docs/git-checkout#git-checkout---patch) which can be applied to a specific file allows for interactive discards of chunks of changes.

**Comment by consideRatio** [comment]
## Using a placeholder in Chart.yaml

We could also use a placeholder in the Chart.yaml's version just like we do for the image tags. I think that makes the most sense. Then, and then we update `chartpress.yaml` to contain the the relevant version field instead.

**Comment by minrk** [comment]
I think `--reset` applying a consistent version to Chart.yaml is a great idea.

**Comment by consideRatio** [comment]
@minrk I considered this and found a catch. The version in Chart.yaml that we want to reset to something like "set-by-chartpress", would cause chartpress normal behavior to malfunction as it actively relies on this version to compose the final version to set.

https://github.com/jupyterhub/chartpress/blob/a79563f600a27de4254648f968b1c0fa6cb06322/chartpress.py#L309-L314

I think it could make sense to start using the latest tag as input instead of Chart.yaml's version. That change would need to go hand in hand with change to `--reset`.

**Comment by consideRatio** [comment]
To get the latest tag made:
```
git for-each-ref refs/tags --sort=-taggerdate --format='%(tag)' --count=1
```

Z2JH latest tags has been:

```
0.8.2
0.8.1
0.8.0
0.8.0-beta.1
0.8.0b1
0.7.0
0.7.0-beta.2
v0.7.0-beta.1
v0.6
v0.5-doc
```

> NOTE: only 0.8.0 were the master branch being tagged, while the 0.8.1 were not tied to a branch, i think... Hmmm... This difference makes `git describe --abbrev=0` show 0.8.0 as compared to 0.8.2.
